src/mem/alloc.h

```diff
@@ -438,8 +598,8 @@ namespace snmalloc
 #else
       handle_message_queue();
 
-      // Free memory of a dynamically known size. Must be called with an
-      // external pointer.
+      p = pagemap().getp(p);
```

Comment:
This is on the fast path? Ensure it is erased on non-Cheri case. Perhaps add a comment?